common/GL/ContextEGL.cpp

```diff
@@ -285,74 +287,54 @@ namespace GL
 		Console.WriteLn(
 			"Trying version %u.%u (%s)", version.major_version, version.minor_version,
 			version.profile == Context::Profile::ES ? "ES" : (version.profile == Context::Profile::Core ? "Core" : "None"));
-		int surface_attribs[16];
-		int nsurface_attribs = 0;
-		surface_attribs[nsurface_attribs++] = EGL_RENDERABLE_TYPE;
-		surface_attribs[nsurface_attribs++] = (version.profile == Profile::ES) ?
-                                                  ((version.major_version >= 3) ? EGL_OPENGL_ES3_BIT :
-                                                                                  ((version.major_version == 2) ? EGL_OPENGL_ES2_BIT : EGL_OPENGL_ES_BIT)) :
-                                                  EGL_OPENGL_BIT;
-		surface_attribs[nsurface_attribs++] = EGL_SURFACE_TYPE;
-		surface_attribs[nsurface_attribs++] = (m_wi.type != WindowInfo::Type::Surfaceless) ? EGL_WINDOW_BIT : 0;
-		surface_attribs[nsurface_attribs++] = EGL_RED_SIZE;
-		surface_attribs[nsurface_attribs++] = 8;
-		surface_attribs[nsurface_attribs++] = EGL_GREEN_SIZE;
-		surface_attribs[nsurface_attribs++] = 8;
-		surface_attribs[nsurface_attribs++] = EGL_BLUE_SIZE;
-		surface_attribs[nsurface_attribs++] = 8;
-		surface_attribs[nsurface_attribs++] = EGL_NONE;
-		surface_attribs[nsurface_attribs++] = 0;
+
+		const auto surface_attribs = [this, version]() -> std::array<int, 16> {
+			const auto opengl_es_version_number = [](auto major_version) {return ((major_version >= 3) ? EGL_OPENGL_ES3_BIT : ((major_version == 2) ? EGL_OPENGL_ES2_BIT : EGL_OPENGL_ES_BIT)); };
+			const auto renderable_type_value = (version.profile == Profile::ES) ? opengl_es_version_number(version.major_version) : EGL_OPENGL_BIT;
+			const auto surface_type_value = (m_wi.type != WindowInfo::Type::Surfaceless) ? EGL_WINDOW_BIT : 0;
+
+			return {EGL_RENDERABLE_TYPE, renderable_type_value, EGL_SURFACE_TYPE, surface_type_value, EGL_RED_SIZE, 8, EGL_GREEN_SIZE, 8, EGL_BLUE_SIZE, 8, EGL_NONE, 0};
+		}();
```

Comment:
_When dealing with unknown code in an unknown codebase, the lambda-form can be a lifesaver in determining and clearly-stating side-effects from unknown external variables!_...But I agree with you @tellowkrinkle , in this case, it is ridiculous now with your clarification! :sweat_smile: Thanks!

---

So... in the original code, the variable [`surface_attribs` was strongly stated as `surface_attribs[16]`](https://github.com/PCSX2/pcsx2/pull/5676/files#diff-5be66afa8049cdaefb8fd5b469f92661ece7f3459cbbfc74f2a1ae3fa6c7e206L288). Because of the `C`-form declaration, the remaining unused array elements were not initialised. This may cause erroneous behaviour and was the main reason why `surface_attribs` was changed to `const std::array<int 16>`.

Is it ok @tellowkrinkle to adjust the datatype you're suggesting (a variable-length `C` array) to `const std::array<int, 16>`? Currently, your form could cause problems if something is expecting an array-datatype with `16` elements...

Reply:
`eglChooseConfig` uses `EGL_NONE` as a terminator, so it should be fine

Reply:
I assume the original choice of 16 was a "I don't want to count these, 16 should be enough" but with an initializer list, the compiler counts for us so we don't have to worry